### DISCUSSION TITLE
Flux cli filter for Ready status

### DISCUSSION THREAD

**Discussion by eloo** [fluxcd/flux2#1578] [Proposals]
Hi,

i would like to suggest a filter function for the flux cli. 
In my current situation i would like to have an option to filter for all flux resources in a specific state.. for example ready==false

So the idea would be to a new cli option like this
```
flux get all --filter=ready=false
```

This would then be a similar syntax as we are already know from kubectl 
```
kubectl get pods --field-selector=status.phase=Running
```

So what do you think? 

Best regards
eloo

**Comment by stefanprodan** [comment]
Maybe we could expose the filter as a flag called `status-selector` where users can specify the status condition name and the desired state e.g.:  `flux get <kind> --status-selector=<condition-name>=<true|false|unknown>` .

This would allow filtering for example Kustomizations with failing health checks:
```
flux get kustomizations --status-selector=healthy=false
```

**Reply by eloo** [reply]
Sounds good 👍
